rotkehlchen/tests/unit/globaldb/test_upgrades.py

```diff
@@ -664,3 +711,274 @@ def test_applying_all_upgrade(globaldb: GlobalDBHandler):
+                    for (collection_id, name, symbol)
+                    in cursor.execute('SELECT id, name, symbol FROM asset_collections')
+                }, cursor.execute(
+                    'SELECT collection_id, asset FROM multiasset_mappings',
+                ).fetchall(),
+            )
+            for cursor in (old_db_cursor, packaged_db_cursor)
+        )
+
+    assets_to_update = []
+    assets_to_package = [
+        f'Asset collection id: {group_id} ({updated_collections[group_id][1]}) not found in packaged DB.'  # noqa: E501
+        for group_id in updated_collections
+        if group_id not in packaged_collections
+    ]
+
+    for group_id in packaged_collections:
+        if group_id not in updated_collections:
+            assert None not in (packaged_collections[group_id][i] for i in (0, 1))
+            assets_to_update.append(ASSET_COLLECTION_UPDATE.format(
+                collection=group_id,
+                name=packaged_collections[group_id][0],
+                symbol=packaged_collections[group_id][1],
+            ))
+            continue
+
+        if packaged_collections[group_id] != updated_collections[group_id]:
+            assets_to_package.append(f'Asset collection id: {group_id} ({packaged_collections[group_id][1]}) - {packaged_collections[group_id]} (packaged) != {updated_collections[group_id]} (asset update)')  # noqa: E501
+
+    assets_to_package.extend([
+        f'Multiasset mapping {mapping} not found in packaged DB.'
+        for mapping in updated_multiasset_mappings
+        if mapping not in packaged_multiasset_mappings
+    ])
+
+    for mapping in packaged_multiasset_mappings:
+        assert None not in (mapping[i] for i in (0, 1))
+        if mapping not in updated_multiasset_mappings:
+            assets_to_update.append(ASSET_MAPPING.format(collection=mapping[0], id=mapping[1]))
+
+    tables = {  # a mapping from EvmTokens field names to the DB table they are stored in
+        'name': 'assets',
+        'address': 'evm_tokens',
+        'chain': 'evm_tokens',
+        'token_kind': 'evm_tokens',
+        'decimals': 'evm_tokens',
+        'protocol': 'evm_tokens',
+        'symbol': 'common_asset_details',
+        'started': 'common_asset_details',
+        'swapped_for': 'common_asset_details',
+        'coingecko': 'common_asset_details',
+        'cryptocompare': 'common_asset_details',
+    }
+
+    # populate mappings from parent token to set of underlying tokens for both DBs
+    updated_underlying_assets: dict[str, set[str]] = defaultdict(set)
+    packaged_underlying_assets: dict[str, set[str]] = defaultdict(set)
+    for underlying_db_assets, underlying_assets in (
+        (updated_underlying_db_assets, updated_underlying_assets),
+        (packaged_underlying_db_assets, packaged_underlying_assets),
+    ):
+        for underlying_identifier, weight, parent_identifier in underlying_db_assets:
+            if float(weight) != 1.0:
```

Comment:
I would make those a FVal since we treat them as so in the backend

```suggestion
            if FVal(weight) != ONE:
```